sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClientEncryptionKey.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.core.util.Context;
+import com.azure.cosmos.implementation.Paths;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyProperties;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyResponse;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.FluxUtil.withContext;
+
+public class CosmosAsyncClientEncryptionKey {
```

Comment:
I wish if we could add this in the encryption project not in the core sdk.

let's discuss in the public api meeting.

Reply:
As discussed , P1 item. Thanks for raising this in meeting aswell